### ISSUE TITLE
Linux installation: cannot verify get.docker.com's certificate

### ISSUE THREAD

**Issue by matm** [docker/docker#14650]
The recommanded installation procedure on Linux with `wget` is failing:
```
wget -qO- https://get.docker.com/ | sh
```
returns no error (exit 0) but nothing happen either.

Trying a simple `wget https://get.docker.com` reveals an SSL certificate verification issue:
```
--2015-07-15 10:44:50--  https://get.docker.com/
Resolving get.docker.com (get.docker.com)... 52.0.250.56, 52.6.66.251, 54.210.203.88
Connecting to get.docker.com (get.docker.com)|52.0.250.56|:443... connected.
ERROR: cannot verify get.docker.com's certificate, issued by `/C=US/O=GeoTrust Inc./CN=RapidSSL SHA256 CA - G3':
  Unable to locally verify the issuer's authority.
To connect to get.docker.com insecurely, use `--no-check-certificate'.
```

So, unfortunately, passing the `--no-check-certificate` option to grab the script is the only way to go at the moment.

**Comment by matm** [comment]
BTW,
```
GNU Wget 1.13.4 built on linux-gnu.
```
does not work but `wget` 1.15 does not complain at all and fetches the script.

**Comment by Evalle** [comment]
Just for the record, with 
$ wget --version
>> GNU Wget 1.16.3 built on linux-gnu  
installation works fine for me.

**Comment by matm** [comment]
Seems like `wget < 1.14` (2012) lacks SNI support (TLS extension added on 1.14+):
- bug report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=653267
- wget 1.14 ChangeLog: http://savannah.gnu.org/forum/forum.php?forum_id=7323

So maybe this version requirement should be stated in the install docs at some point.

**Comment by matm** [comment]
For example,

Ubuntu `12.04.5 LTS` is supported until April 2017 and comes with `wget` 1.13.4.

**Comment by cpuguy83** [comment]
I don't feel like a change is needed, other than to use curl in the docs, b/c curl is of course superior in every way :)

cc @moxiegirl

**Comment by moxiegirl** [comment]
@cpuguy83 So, we went round on this a bit in email way back and came up with this: https://github.com/docker/docker/issues/11023 and ended up with wget because Google. (Also this https://github.com/docker/docker/issues/7422_

Maybe I should just show both wget/curl and let folks decide?

**Comment by matm** [comment]
@moxiegirl yep. And if we keep the `wget` command in the install docs, I think a note on minimum required version should be appended.

**Comment by ankushagarwal** [comment]
Hey @moxiegirl, I went through #11023 and #7422 and did not find a compelling reason to use curl over wget. I agree with @cpuguy83, we should use curl because of the wget versioning issue.

`wget -qO- https://get.docker.com/ | sh `   -->   `curl -sSL https://get.docker.com/  | sh`
 
I'll be happy to update the docs if we can agree on it.

**Comment by moxiegirl** [comment]
@ankushagarwal this move to `wget` seems to cause more issues than the original so I'm all for `curl` if there is consensus there.  Have at it @ankushagarwal thanks for the review.

**Comment by ankushagarwal** [comment]
Tagging infrastructure maintainers for feedback and consensus : @jfrazelle @crosbymichael 

TLDR : We want to change install docs to use `curl` instead of `wget` because wget lacks SNI support for versions < 1.14 and hence installation fails because it cannot verify get.docker.com's certificate.

**Comment by jfrazelle** [comment]
I like curl better than wget, seems fine to me.

**Comment by aaronmarzi** [comment]
https://docs.docker.com/linux/step_one/ is not updated yet

**Comment by thaJeztah** [comment]
@aaronmarzi I opened a pull request here: https://github.com/docker/tutorials/issues/68
